pkg/loop/plugin_median.go

```diff
@@ -37,116 +39,111 @@ func PluginMedianHandshakeConfig() plugin.HandshakeConfig {
 	}
 }
 
-func PluginMedianClientConfig(lggr logger.Logger) *plugin.ClientConfig {
-	return &plugin.ClientConfig{
-		HandshakeConfig: PluginMedianHandshakeConfig(),
-		Plugins: map[string]plugin.Plugin{
-			PluginMedianName: NewGRPCPluginMedian(nil, lggr),
-		},
-		AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
-	}
-}
-
-type grpcPluginMedian struct {
+type GRPCPluginMedian struct {
 	plugin.NetRPCUnsupportedPlugin
 
-	impl PluginMedian
-	lggr logger.Logger
-}
+	StopCh <-chan struct{}
+	Logger logger.Logger
 
-func NewGRPCPluginMedian(mp PluginMedian, lggr logger.Logger) plugin.Plugin {
-	return &grpcPluginMedian{impl: mp, lggr: lggr}
+	PluginServer PluginMedian
+
+	pluginClient *pluginMedianClient
```

Comment:
please use a different name for this field. we use `pluginClient` as a struct name in this package to mean something different that this (right? this is not 1:1 with `pluginClient` defn'd in plugin.go`? it's more and other stuff)

Reply:
right now we have pluginClient := &pluginMedianClient{*pluginClient} which i can't follow

Reply:
What do you mean? The struct name and the field type give the extra context. I even tried to name these just `Server` and `client`, but that clashes with the net rpc  method names. I think saying more would just be redundant no? What do you propose?

There might be a way to simplify this more broadly with generics, but IIRC I already tried that once. e.g. `GRPCPlugin[P ...]`